### PR TITLE
[ADD] new module "partner_refunds_buttons"

### PR DESCRIPTION
**Pull request by anthony-muschang**
This module add new smart buttons in the Partner view to Customer Refunds and Supplier Refunds.

### CODE REVIEW

File: account_refund_partner/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+# Author: Anthony Muschang
+# Copyright (c) 2014 Acsone SA/NV (http://www.acsone.eu)
+# All Rights Reserved
```
Comment (sbidoul) [inline account_refund_partner/__openerp__.py]: Remove All Rights Reserved

File: account_refund_partner/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+#
+# WARNING: This program as such is intended to be used by professional
```
Comment (sbidoul) [inline account_refund_partner/__openerp__.py]: Remove this non-standard license section

File: account_refund_partner/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    'category': 'Accounting & Finance',
+    'license': 'AGPL-3',
+    'description': """
+Add shortcuts to Refunds in partner screen
+==========================================
+
+This module add shortcuts (i.e: smart button) in the Partner view to
+ Customer Refunds and Supplier Refunds
+""",
+    'author': 'Anthony Muschang, ACSONE',
```
Comment (sbidoul) [inline account_refund_partner/__openerp__.py]: author: ACSONE SA/NV for consistency with our other modules

File: account_refund_partner/tests/__init__.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+#
+##############################################################################
+
+from . import test_refund_count
+
+fast_suite = [
```
Comment (sbidoul) [inline account_refund_partner/tests/__init__.py]: fast_suite and checks are not necessary in 8.0

File: account_refund_partner/res_partner.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+##############################################################################
+from openerp import models, fields, api
+
+
+class res_partner(models.Model):
+    _inherit = "res.partner"
+
+    supplier_refund_count = fields.Integer(compute='_refund_count',
+                                           string="# Supplier Refund",
+                                           groups='account.' +
```
Comment (pedrobaeza) [inline account_refund_partner/res_partner.py]: You don't need to put `+` sign. Consecutive strings are concatenated automatically, but it's better to use this indentation (also PEP8 compliant) in these cases:

```
supplier_refund_count = fields.Integer(
    compute='_refund_count', string="# Supplier Refund", 
    groups='account.group_account_invoice')
```

File: account_refund_partner/res_partner.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+                                           'group_account_invoice')
+
+    customer_refund_count = fields.Integer(compute='_refund_count',
+                                           string="# Supplier Refund",
+                                           groups='account.' +
+                                           'group_account_invoice')
+
+    @api.multi
```
Comment (pedrobaeza) [inline account_refund_partner/res_partner.py]: Put @api.one to avoid one extra indentation level and an unneeded `for`.

**Comment by sbidoul** [comment]
I'm not easy with the module name. Is account_partner_refund_buttons more intuitive?

**Comment by pedrobaeza** [comment]
I would say _partner_refunds_buttons_ to follow convention about first word is the area affected.

**Comment by sbidoul** [comment]
@pedrobaeza good.

**Comment by anthony-muschang** [comment]
Thanks for the feedback. I renamed the module and made some improvements.

**Comment by pedrobaeza** [comment]
Thanks for the changes. One doubt: standard smart button 'Invoices' includes also refunds in its domain? If so, maybe we should modify it to extract that part, because these buttons make the trick.

**Comment by sbidoul** [comment]
@pedrobaeza you are right. The Invoiced button includes refunds. But when you click on it, it shows a mix of invoices and refunds, with a wrong total. It even mixes supplier and customer invoices if the partner has both! This looks like a bug to me.

@anthony-muschang I suggest the following:
- put the Sup. Refund button next to the Sup. Invoices button
- add a Cust. Invoices button with a count before your Cust. Refund button.

In parallel, we must see with Odoo what to do with this weird Invoiced button.

**Comment by anthony-muschang** [comment]
- I cannot put the 'Sup. Refund' button next to the "Sup. Invoices" without adding the module purchase to this module dependencies.
- Ok for adding the 'Cust. Invoices' button. 

For information the "Invoiced" button is added by account and the "Sup. Invoices" button is added by purchase.

**Comment by pedrobaeza** [comment]
You can put it without depending on purchase, but not modifying the other button.

**Comment by sbidoul** [comment]
@pedrobaeza do you think this module makes sense for OCA?

We'll discuss internally, but I think I've been too optimistic. Since the standard buttons are not very coherent, it's difficult to make something clean.

**Comment by pedrobaeza** [comment]
Yeah, maybe you're right. We should push Odoo to include it in core.

**Comment by legalsylvain** [comment]
I close this PR, due to last old comments.

**Pull request closed without merging**